packages/base/src/features/patchPopup.ts

```diff
@@ -68,16 +79,56 @@ const hasWebComponentPopupAbove = (popup: object) => {
 	return false;
 };
 
-const openNativePopover = (domRef: HTMLElement) => {
+const enableNativePopoverForOpenUI5 = (domRef: HTMLElement, popup: OpenUI5Popup) => {
+	const openUI5BlockLayer = document.getElementById("sap-ui-blocklayer-popup");
+
+	if (popup.getModal() && openUI5BlockLayer) {
+		openUI5BlockLayer.setAttribute("popover", "manual");
+		openUI5BlockLayer.hidePopover();
+		openUI5BlockLayer.showPopover();
+	}
+
 	domRef.setAttribute("popover", "manual");
 	domRef.showPopover();
 };
 
-const closeNativePopover = (domRef: HTMLElement) => {
+const disableNativePopoverForOpenUI5 = (domRef: HTMLElement) => {
 	if (domRef.hasAttribute("popover")) {
 		domRef.hidePopover();
 		domRef.removeAttribute("popover");
 	}
+
+	const lastPopup = AllOpenedPopupsRegistry.openedRegistry[AllOpenedPopupsRegistry.openedRegistry.length - 1];
+	if (lastPopup.type === "OpenUI5" && lastPopup.instance.getModal()) {
+		const openUI5BlockLayer = document.getElementById("sap-ui-blocklayer-popup");
+		if (openUI5BlockLayer && openUI5BlockLayer.hasAttribute("popover")) {
+			openUI5BlockLayer.hidePopover();
+		}
+	}
+};
+
+const fixOpenUI5PopupBelow = () => {
+	if (!isNativePopoverOpen()) {
+		return;
+	}
+
+	const prevPopup = AllOpenedPopupsRegistry.openedRegistry[AllOpenedPopupsRegistry.openedRegistry.length - 2];
+	if (!prevPopup || prevPopup.type !== "OpenUI5" || !prevPopup.instance.getModal()) {
+		return;
+	}
+
+	const prevPopupContent = prevPopup.instance.getContent()!;
+	const content = prevPopupContent instanceof HTMLElement ? prevPopupContent : prevPopupContent?.getDomRef();
```

Comment:
now you have this 2 rows on 3 places. Make it a separate funciton. E.g. getPopupContentElement